learn/src/main/java/org/navgurukul/learn/ui/learn/viewholder/AssessmentFragmentViewModel.kt

```diff
@@ -99,22 +107,34 @@ class AssessmentFragmentViewModel (
         updateListInLocalDb(currentStateList)
     }
 
-    private fun showCorrectOnIncorrect(){
+    private fun showCorrectOnIncorrect() {
         viewModelScope.launch {
-            val correctOption = (allAssessmentContentList
-                .find { it.component == BaseCourseContent.COMPONENT_SOLUTION } as SolutionBaseCourseContent)
-                .value
             val currentState = viewState.value!!
+            val solutionContent = (allAssessmentContentList
+                .find { it.component == BaseCourseContent.COMPONENT_SOLUTION } as SolutionBaseCourseContent)
+            val correctOptions = solutionContent.correct_options_value
+            Log.d("correctOptions", correctOptions.toString())
+            val incorrectOptions = solutionContent.incorrect_options_value
+            Log.d("incorrectOptions", incorrectOptions.toString())
+
             currentState.assessmentContentListForUI.forEach {
-                if (it.component == BaseCourseContent.COMPONENT_OPTIONS){
+                if (it.component == BaseCourseContent.COMPONENT_OPTIONS) {
                     val optionList = it as OptionsBaseCourseContent
-                    for (option in optionList.value){
-                        if (option.id == correctOption){
-                            option.viewState = OptionViewState.CORRECT
-                        } else if (option.id == selectedOption){
-                            option.viewState = OptionViewState.INCORRECT
-                        }else {
-                            option.viewState = OptionViewState.NOT_SELECTED
+                    for (option in optionList.value) {
+                        Log.d("option", option.id.toString())
+                        if (option.id in correctOptions.map { it.value }) {
+                            if(option.id in selectedOption) {
+                                option.viewState = OptionViewState.CORRECT
+                            }
+                            /* 1 -> modify same function to take only selected or unselected values
+                            *  2 -> modify same function to take PARTIALLY_CORRECT and PARTIALLY_INCORRECT values
+                            * */
+//                            option.viewState = OptionViewState.CORRECT
+                            //Log.d("option", option.viewState.toString())
+                        } else if (option.id in incorrectOptions!!.map { it.value }) {
```

Comment:
let's try to avoid usage of !! operator, instead use ? fo null safety along with scoped variables like .let or just provide a default value in case of null by ?: operator.